test/Window/WindowHandleWin32.test.cpp

```diff
@@ -0,0 +1,143 @@
+#include <SFML/Window/WindowHandle.hpp>
+
+// Other 1st party headers
+#include <SFML/Window/Window.hpp>
+#include <SFML/Window/WindowBase.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <SystemUtil.hpp>
+#include <optional>
+
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#include <Windows.h>
+
+namespace
+{
+bool gotWmShowWindow = false;
+
+LRESULT WINAPI wndProc(HWND handle, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    switch (message)
+    {
+        case WM_SHOWWINDOW:
+            gotWmShowWindow = true;
+            assert(wParam == TRUE); // If wParam is TRUE, the window is being shown
+            assert(lParam == 0); // If lParam is zero, the message was sent because of a call to the ShowWindow function
+            break;
+        case WM_CLOSE:
+            PostQuitMessage(0);
+            return 0;
+    }
+    return DefWindowProcW(handle, message, wParam, lParam);
+}
+} // namespace
+
+TEST_CASE("[Window] sf::WindowHandle (Win32)")
+{
+    const WNDCLASSW classInfo{{}, wndProc, {}, {}, GetModuleHandleW(nullptr), {}, {}, {}, {}, L"sfml_WindowHandleTests"};
+
+    const ATOM winClassId = RegisterClassW(&classInfo);
+    REQUIRE(winClassId);
+
+    const sf::WindowHandle handle = CreateWindowW(reinterpret_cast<LPWSTR>(static_cast<ULONG_PTR>(winClassId)),
```

Comment:
It would be interesting to have another case with `CreateWindowExW`, and passing it the `WS_EX_TOOLWINDOW` flag.
This results in a different window border, that sfml might not handle correctly yet.
Another thing would be adding a native menu to the Window, this is also something that is not handled correctly yet.

Reply:
I like these ideas. Let's explore this in a future PR.